src/richie/apps/courses/models/course.py

```diff
@@ -482,6 +482,37 @@ def course_runs_enrollment_count(self):
             direct_course__extended_object__publisher_is_draft=is_draft,
         ).aggregate(sum=Sum("enrollment_count"))["sum"]
 
+    @cached_property
+    def course_languages_display(self):
+        """
+        Returns a string listing of languages available for this course across all its
+        course runs (merged).
+        They may be directly related to the course or to a snapshot of the course.
+
+        The draft and the public page have their own course runs. The course runs on the draft
+        page are copied to the public page when the course is published (for any language).
+        """
+        is_draft = self.extended_object.publisher_is_draft
+        node = self.extended_object.node
+        current_and_descendant_nodes = node.__class__.get_tree(parent=node)
+
+        course_runs = (
+            CourseRun.objects.filter(
+                direct_course__extended_object__node__in=current_and_descendant_nodes,
+                direct_course__extended_object__publisher_is_draft=is_draft,
+            )
+            .exclude(catalog_visibility="hidden")
+            .only("languages")
+        )
+        languages = list(
+            {x for course_languages in course_runs for x in course_languages.languages}
+        )
+
+        languages.sort()
```

Comment:
Actually this is sorting on the machine name instead of the language name.

I have fixed this while testing the PR and added more tests to secure how it works: https://github.com/openfun/richie/commit/f4b8d80c0fa355010b08b623ba6d2bf292eb8bce

Reply:
Ok, thank you!